scripts/install_apt_deps.sh

```diff
@@ -1,21 +1,252 @@
+        if [ -z "${key_id}" ]; then
+            curl "${key_srv_url}" | sudo apt-key add -
+        else
+            sudo apt-key adv --keyserver "${key_srv_url}" --recv-key "${key_id}"
+        fi
+        if [ "${?}" -eq "0" ]; then
+            break # Stop adding keys after the first success
+        fi
+    done
+    sudo add-apt-repository -u -y "deb ${repo_url} ${VERSION_CODENAME} ${repo_comp}"
+}
+
+# Generate the list of package dependencies to install based on the provided package groups
+# The indirect variables referenced by the group names should be bash arrays
+# Good call site: list=$(pkg_dep_list_from "${group_list}")
+# Bad call site:  list="$(pkg_dep_list_from ${group_list})"
+function pkg_dep_list_from() {
+    local pkg_dep_list=""
+    local pkg_dep_groups=${1}
+    for group in ${pkg_dep_groups}; do
+        pkg_dep_list_group_var="pkg_dep_list_${group}"
+        pkg_dep_list_group="${pkg_dep_list_group_var}[@]"
+        pkg_dep_list+="${!pkg_dep_list_group} "
+    done
+    echo "${pkg_dep_list}"
+}
+
+
+### Package dependencies setup ###
+
+# List of common package dependencies for the ILLIXR project,
+# grouped by purpose/feature
+
+pkg_dep_list_prereq=(
+    curl
+    gnupg2
+    software-properties-common
+) # End list
+
+pkg_dep_list_common=(
+    build-essential
+    git
+    clang-10
+    make
+    cmake
+    unzip
+    wget
+    pkg-config
+) # End list
+
+pkg_dep_list_gl=(
+    glew-utils
+    glslang-tools
+    freeglut3-dev
+    libglew-dev
+    libglfw3-dev
+    libvirglrenderer-dev
+) # End list
+
+pkg_dep_list_mesa=(
+    mesa-common-dev
+    libglu1-mesa-dev
+    libdrm-dev
+) # End list
+
+pkg_dep_list_display=(
+    xvfb
+    libx11-xcb-dev
+    libx11-dev
+    libxcb-glx0-dev
+    libxcb-randr0-dev
+    libxrandr-dev
+    libxkbcommon-dev
+    libwayland-dev
+) # End list
+
+pkg_dep_list_image=(
+    libjpeg-dev
+    libpng-dev
+    libtiff-dev
+    libvtk6-dev
+) # End list
+
+pkg_dep_list_sound=(
+    libpulse-dev
+) # End list
+
+pkg_dep_list_usb=(
+    libusb-dev
+    libusb-1.0
+    libudev-dev
+    libv4l-dev
+    libhidapi-dev
+) # End list
+
+pkg_dep_list_thread=(
+    libc++-dev
+    libc++abi-dev
+    libboost-all-dev
+    libtbb-dev
+) # End list
+
+pkg_dep_list_math=(
+    gfortran
+    libeigen3-dev
+    libblas-dev
+    libsuitesparse-dev
+    libparmetis-dev
+    libatlas-base-dev
+) # End list
 
-sudo apt install -y software-properties-common curl gnupg2
-curl https://apt.kitware.com/keys/kitware-archive-latest.asc | sudo apt-key add -
-sudo apt-add-repository -u -y "deb https://apt.kitware.com/ubuntu/ ${UBUNTU_CODENAME} main"
-sudo apt-key adv --keyserver keys.gnupg.net --recv-key F6E65AC044F831AC80A06380C8B3A55A6F3EFCDE || sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-key F6E65AC044F831AC80A06380C8B3A55A6F3EFCDE
-sudo add-apt-repository "deb http://realsense-hw-public.s3.amazonaws.com/Debian/apt-repo bionic main" -u
+pkg_dep_list_nogroup=(
```

Comment:
@mhuzai @charmoniumQ should we group these apt packages by the plugins that need them? May help with debugging.

Reply:
Looks like mkdocs is never installed by the scripts. Seems like a dependency only required for ILLIXR devs, so I think if there are many of these types of packages eventually, we can add a group for them that gets installed with a dev flag.

Reply:
Yeah, currently we only manually install it, but I like the idea of a dev flag for install deps.